asciidoc-confluence-publisher-client/src/main/java/org/sahli/asciidoc/confluence/publisher/client/ConfluencePublisher.java

```diff
@@ -67,16 +70,39 @@ public void publish() {
         assertMandatoryParameter(isNotBlank(this.metadata.getSpaceKey()), "spaceKey");
         assertMandatoryParameter(isNotBlank(this.metadata.getAncestorId()), "ancestorId");
 
-        startPublishingUnderAncestorId(this.metadata.getPages(), this.metadata.getSpaceKey(), this.metadata.getAncestorId());
+        switch(this.metadata.getPublishStrategy()) {
+            case APPEND_TO_ANCESTOR:
+                startPublishingUnderAncestorId(this.metadata.getPages(), this.metadata.getSpaceKey(), this.metadata.getAncestorId());
+                break;
+            case REPLACE_ANCESTOR:
+                // verify that only a single root exists
+                if (this.metadata.getPages().size() > 1) {
+                    throw new IllegalArgumentException(String.format("Multiple root pages detected: %s. " +
+                        "Publishing to confluence with the %s strategy only allows a single root to be defined.",
+                        StringUtils.join(this.metadata.getPages().stream().map(page -> "'" + page.getTitle() + "'").collect(Collectors.toList()), ", "),
```

Comment:
We tried to avoid using external libraries as much as possible, especially for such simple functionality. In this case, `this.metadata.getPages().stream().map(page -> "'" + page.getTitle() + "'").collect(joining(", "))` would do the trick using jdk api.